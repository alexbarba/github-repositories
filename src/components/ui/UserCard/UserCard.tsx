import { FC } from 'react'
import { IUser } from 'types'
import { Avatar } from 'components/ui'
import {
  LinkIcon,
  PeopleIcon,
  StarIcon,
  MailIcon,
  LocationIcon,
  OrganizationIcon
} from '@primer/octicons-react'

type Props = {
  user: IUser
}

const UserCard: FC<Props> = ({ user }) => {
  const Social: FC<{ className?: string }> = ({ className }) => (
    <div className={className}>
      <a href={`${user.html_url}?tab=followers`}>
        <PeopleIcon size={16} className="mr-2" />
        <span className="font-bold text-primary">{user.followers}</span>{' '}
        followers
      </a>{' '}
      ·{' '}
      <a href={`${user.html_url}?tab=following`}>
        <span className="font-bold text-primary">{user.following}</span>{' '}
        following{' '}
      </a>
      ·{' '}
      <a href={`${user.html_url}?tab=stars`}>
        <StarIcon size={16} />{' '}
        <span className="font-bold text-primary">38</span>
      </a>
    </div>
  )

  return (
    <div className="w-full text-sm text-primary">
      <div className="flex items-center md:block">
        <Avatar
          src={user.avatar_url}
          className="relative inline-block items-center h-auto w-2/12 md:w-full py-4 mr-3 md:mx-0 md:py-0 min-w-[3rem] min-h-[3rem]"
        />
        <div className="inline-block w-10/12 py-4 md:w-full">
          <h2 className="text-2xl font-bold">{user.name}</h2>
          <h3 className="text-lg text-secondary">{user.login}</h3>
        </div>
      </div>
      <button className="hidden btn md:block">Follow</button>
      <p className="my-4 text-base">{user.bio}</p>
      <div>
        <Social className="hidden mb-3 text-secondary md:block" />

        {user.company ? (
          <p className="hidden md:block pt-[5px]">
            <OrganizationIcon
              size={16}
              className="hidden mr-2 text-secondary"
            />
            {user.company}
          </p>
        ) : null}
        {user.location ? (
          <p className="hidden md:block">
            <LocationIcon size={16} className="mr-2 text-secondary" />
            {user.location}
          </p>
        ) : null}
        <p className="pt-[5px]">
          <MailIcon size={16} className="mr-2 text-secondary" />
          {user.email || 'mail@example.com'}
        </p>
        {user.blog ? (
          <p className="pt-[5px]">
            <LinkIcon size={16} className="mr-2 text-secondary" />
            {user.blog}
          </p>
        ) : null}
      </div>
      <Social className="block mt-2 mb-4 md:my-4 text-secondary md:hidden" />
      <button className="btn md:hidden">Follow</button>
    </div>
  )
}

export default UserCard
