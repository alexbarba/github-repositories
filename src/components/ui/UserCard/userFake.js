export default {
  login: 'yknx4',
  id: 1848186,
  node_id: 'MDQ6VXNlcjE4NDgxODY=',
  avatar_url: 'https://avatars.githubusercontent.com/u/1848186?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/yknx4',
  html_url: 'https://github.com/yknx4',
  followers_url: 'https://api.github.com/users/yknx4/followers',
  following_url: 'https://api.github.com/users/yknx4/following{/other_user}',
  gists_url: 'https://api.github.com/users/yknx4/gists{/gist_id}',
  starred_url: 'https://api.github.com/users/yknx4/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/yknx4/subscriptions',
  organizations_url: 'https://api.github.com/users/yknx4/orgs',
  repos_url: 'https://api.github.com/users/yknx4/repos',
  events_url: 'https://api.github.com/users/yknx4/events{/privacy}',
  received_events_url: 'https://api.github.com/users/yknx4/received_events',
  type: 'User',
  site_admin: false,
  name: 'Ale Ornelas Figueroa',
  company: '@HeyHomie',
  blog: 'http://www.ale.world',
  location: 'Mexico',
  email: null,
  hireable: true,
  bio: 'Devops Leader at @HeyHomie\r\n\r\nFormer Devops Engineer at @gitamv',
  twitter_username: null,
  public_repos: 190,
  public_gists: 31,
  followers: 36,
  following: 9,
  created_at: '2012-06-13T21:56:55Z',
  updated_at: '2021-12-01T23:01:43Z'
}
