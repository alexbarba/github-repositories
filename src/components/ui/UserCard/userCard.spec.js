import { render } from '@testing-library/react'
import UserCard from './UserCard'
import userFake from './userFake'

describe('UserCard', () => {
  it('should be match with snapshot', () => {
    const { asFragment } = render(<UserCard user={userFake} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
