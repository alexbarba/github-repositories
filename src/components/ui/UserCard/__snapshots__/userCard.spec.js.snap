// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should be match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full text-sm text-primary"
  >
    <div
      class="flex items-center md:block"
    >
      <figure
        class="relative inline-block items-center h-auto w-2/12 md:w-full py-4 mr-3 md:mx-0 md:py-0 min-w-[3rem] min-h-[3rem]"
      >
        <a
          href="https://avatars.githubusercontent.com/u/1848186?v=4"
        >
          <img
            alt="avatar"
            class="w-full h-auto rounded-full"
            height="296"
            src="https://avatars.githubusercontent.com/u/1848186?v=4"
            width="296"
          />
        </a>
      </figure>
      <div
        class="inline-block w-10/12 py-4 md:w-full"
      >
        <h2
          class="text-2xl font-bold"
        >
          Ale Ornelas Figueroa
        </h2>
        <h3
          class="text-lg text-secondary"
        >
          yknx4
        </h3>
      </div>
    </div>
    <button
      class="hidden btn md:block"
    >
      Follow
    </button>
    <p
      class="my-4 text-base"
    >
      Devops Leader at @HeyHomie

Former Devops Engineer at @gitamv
    </p>
    <div>
      <div
        class="hidden mb-3 text-secondary md:block"
      >
        <a
          href="https://github.com/yknx4?tab=followers"
        >
          <svg
            aria-hidden="true"
            class="mr-2"
            fill="currentColor"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M5.5 3.5a2 2 0 100 4 2 2 0 000-4zM2 5.5a3.5 3.5 0 115.898 2.549 5.507 5.507 0 013.034 4.084.75.75 0 11-1.482.235 4.001 4.001 0 00-7.9 0 .75.75 0 01-1.482-.236A5.507 5.507 0 013.102 8.05 3.49 3.49 0 012 5.5zM11 4a.75.75 0 100 1.5 1.5 1.5 0 01.666 2.844.75.75 0 00-.416.672v.352a.75.75 0 00.574.73c1.2.289 2.162 1.2 2.522 2.372a.75.75 0 101.434-.44 5.01 5.01 0 00-2.56-3.012A3 3 0 0011 4z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="font-bold text-primary"
          >
            36
          </span>
           followers
        </a>
         路 
        <a
          href="https://github.com/yknx4?tab=following"
        >
          <span
            class="font-bold text-primary"
          >
            9
          </span>
           following 
        </a>
        路 
        <a
          href="https://github.com/yknx4?tab=stars"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-star"
            fill="currentColor"
            height="16"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
              fill-rule="evenodd"
            />
          </svg>
           
          <span
            class="font-bold text-primary"
          >
            38
          </span>
        </a>
      </div>
      <p
        class="hidden md:block pt-[5px]"
      >
        <svg
          aria-hidden="true"
          class="hidden mr-2 text-secondary"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M1.5 14.25c0 .138.112.25.25.25H4v-1.25a.75.75 0 01.75-.75h2.5a.75.75 0 01.75.75v1.25h2.25a.25.25 0 00.25-.25V1.75a.25.25 0 00-.25-.25h-8.5a.25.25 0 00-.25.25v12.5zM1.75 16A1.75 1.75 0 010 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 00.25-.25V8.285a.25.25 0 00-.111-.208l-1.055-.703a.75.75 0 11.832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0114.25 16h-3.5a.75.75 0 01-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 01-.75-.75V14h-1v1.25a.75.75 0 01-.75.75h-3zM3 3.75A.75.75 0 013.75 3h.5a.75.75 0 010 1.5h-.5A.75.75 0 013 3.75zM3.75 6a.75.75 0 000 1.5h.5a.75.75 0 000-1.5h-.5zM3 9.75A.75.75 0 013.75 9h.5a.75.75 0 010 1.5h-.5A.75.75 0 013 9.75zM7.75 9a.75.75 0 000 1.5h.5a.75.75 0 000-1.5h-.5zM7 6.75A.75.75 0 017.75 6h.5a.75.75 0 010 1.5h-.5A.75.75 0 017 6.75zM7.75 3a.75.75 0 000 1.5h.5a.75.75 0 000-1.5h-.5z"
            fill-rule="evenodd"
          />
        </svg>
        @HeyHomie
      </p>
      <p
        class="hidden md:block"
      >
        <svg
          aria-hidden="true"
          class="mr-2 text-secondary"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M11.536 3.464a5 5 0 010 7.072L8 14.07l-3.536-3.535a5 5 0 117.072-7.072v.001zm1.06 8.132a6.5 6.5 0 10-9.192 0l3.535 3.536a1.5 1.5 0 002.122 0l3.535-3.536zM8 9a2 2 0 100-4 2 2 0 000 4z"
            fill-rule="evenodd"
          />
        </svg>
        Mexico
      </p>
      <p
        class="pt-[5px]"
      >
        <svg
          aria-hidden="true"
          class="mr-2 text-secondary"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M1.75 2A1.75 1.75 0 000 3.75v.736a.75.75 0 000 .027v7.737C0 13.216.784 14 1.75 14h12.5A1.75 1.75 0 0016 12.25v-8.5A1.75 1.75 0 0014.25 2H1.75zM14.5 4.07v-.32a.25.25 0 00-.25-.25H1.75a.25.25 0 00-.25.25v.32L8 7.88l6.5-3.81zm-13 1.74v6.441c0 .138.112.25.25.25h12.5a.25.25 0 00.25-.25V5.809L8.38 9.397a.75.75 0 01-.76 0L1.5 5.809z"
            fill-rule="evenodd"
          />
        </svg>
        mail@example.com
      </p>
      <p
        class="pt-[5px]"
      >
        <svg
          aria-hidden="true"
          class="mr-2 text-secondary"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z"
            fill-rule="evenodd"
          />
        </svg>
        http://www.ale.world
      </p>
    </div>
    <div
      class="block mt-2 mb-4 md:my-4 text-secondary md:hidden"
    >
      <a
        href="https://github.com/yknx4?tab=followers"
      >
        <svg
          aria-hidden="true"
          class="mr-2"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M5.5 3.5a2 2 0 100 4 2 2 0 000-4zM2 5.5a3.5 3.5 0 115.898 2.549 5.507 5.507 0 013.034 4.084.75.75 0 11-1.482.235 4.001 4.001 0 00-7.9 0 .75.75 0 01-1.482-.236A5.507 5.507 0 013.102 8.05 3.49 3.49 0 012 5.5zM11 4a.75.75 0 100 1.5 1.5 1.5 0 01.666 2.844.75.75 0 00-.416.672v.352a.75.75 0 00.574.73c1.2.289 2.162 1.2 2.522 2.372a.75.75 0 101.434-.44 5.01 5.01 0 00-2.56-3.012A3 3 0 0011 4z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="font-bold text-primary"
        >
          36
        </span>
         followers
      </a>
       路 
      <a
        href="https://github.com/yknx4?tab=following"
      >
        <span
          class="font-bold text-primary"
        >
          9
        </span>
         following 
      </a>
      路 
      <a
        href="https://github.com/yknx4?tab=stars"
      >
        <svg
          aria-hidden="true"
          class="octicon octicon-star"
          fill="currentColor"
          height="16"
          role="img"
          style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
            fill-rule="evenodd"
          />
        </svg>
         
        <span
          class="font-bold text-primary"
        >
          38
        </span>
      </a>
    </div>
    <button
      class="btn md:hidden"
    >
      Follow
    </button>
  </div>
</DocumentFragment>
`;
